tests/unit/modules/tls_test.py

```diff
@@ -0,0 +1,623 @@
+            'fingerprint': ('96:72:B3:0A:1D:34:37:05:75:57:44:7E:08:81:A7:09:'
+                            '0C:E1:8F:5F:4D:0C:49:CE:5B:D2:6B:45:D3:4D:FF:31'),
+            'serial_number': 284092004844685647925744086791559203700L,
+            'subject': {
+                'C': 'US',
+                'CN': 'localhost',
+                'L': 'Salt Lake City',
+                'O': 'SaltStack',
+                'ST': 'Utah',
+                'emailAddress':
+                'xyz@pdq.net'},
+            'not_before': 1430843961.0,
+            'issuer': {
+                'C': 'US',
+                'CN': 'localhost',
+                'L': 'Salt Lake City',
+                'O': 'SaltStack',
+                'ST': 'Utah',
+                'emailAddress': 'xyz@pdq.net'}
+        }
+
+        def ignore_extensions(data):
+            '''
+            Ignore extensions pending a resolution of issue 24338
+            '''
+            if 'extensions' in data.keys():
+                data['extensions'] = None
+            return data
+
+        # older pyopenssl versions don't have extensions or
+        # signature_algorithms
+        def remove_not_in_result(source, reference):
+            if 'signature_algorithm' not in reference:
+                del source['signature_algorithm']
+            if 'extensions' not in reference:
+                del source['extensions']
+        result = ignore_extensions(tls.cert_info(certp))
+        remove_not_in_result(ret, result)
+        self.assertEqual(result, ret)
+
+    @patch('salt.modules.tls.maybe_fix_ssl_version',
+           MagicMock(return_value=True))
+    def test_create_ca(self):
+        '''
+        Test creating CA cert
+        '''
+        ca_path = tempfile.mkdtemp(dir=integration.SYS_TMP_DIR)
+        try:
+            ca_name = 'test_ca'
+            certp = '{0}/{1}/{2}_ca_cert.crt'.format(
+                ca_path,
+                ca_name,
+                ca_name)
+            certk = '{0}/{1}/{2}_ca_cert.key'.format(
+                ca_path,
+                ca_name,
+                ca_name)
+            ret = 'Created Private Key: "{0}." Created CA "{1}": "{2}."'.format(
+                certk, ca_name, certp)
+            mock = MagicMock(return_value=ca_path)
+            with patch.dict(tls.__salt__, {'config.option': mock}):
+                with patch.dict(tls.__opts__, {'hash_type': 'sha256',
+                                               'cachedir': ca_path}):
+                    self.assertEqual(
+                        tls.create_ca(
+                            ca_name,
+                            days=365,
+                            fixmode=False,
+                            **_TLS_TEST_DATA['create_ca']),
+                        ret)
+        finally:
+            if os.path.isdir(ca_path):
+                shutil.rmtree(ca_path)
+
+    @patch('salt.modules.tls.maybe_fix_ssl_version',
+           MagicMock(return_value=True))
+    def test_recreate_ca(self):
```

Comment:
This needs a `@destructiveTest` (`from salttesting.helpers import destructiveTest`).